src/main/java/twilightforest/block/BlockTFCicada.java

```diff
@@ -11,11 +18,73 @@ public class BlockTFCicada extends BlockTFCritter {
 
     protected BlockTFCicada() {
         super();
+        this.dropItem = Items.dye;
+        this.dropMeta = 2;
     }
 
     @Override
     public TileEntity createTileEntity(World world, int metadata) {
         return new TileEntityTFCicada();
     }
 
+    /**
+     * Triggered whenever an entity collides with this block (enters into the block). Args: world, x, y, z, entity
+     */
+    @Override
+    public void onEntityCollidedWithBlock(World world, int x, int y, int z, Entity entity) {
+        super.onEntityCollidedWithBlock(world, x, y, z, entity);
+        // stopSinging(worldIn, x, y, z);
+    }
+
+    @Override
+    public boolean dropCritterIfCantStay(World world, int x, int y, int z) {
+        stopSinging(world, x, y, z);
+        return super.dropCritterIfCantStay(world, x, y, z);
+    }
+
+    /**
+     * Called when the block is attempted to be harvested
+     */
+    public void onBlockHarvested(World worldIn, int x, int y, int z, int meta, EntityPlayer player) {
+        stopSinging(worldIn, x, y, z);
+    }
+
+    /**
+     * Called right before the block is destroyed by a player. Args: world, x, y, z, metaData
+     */
+    public void onBlockDestroyedByPlayer(World worldIn, int x, int y, int z, int meta) {
+        stopSinging(worldIn, x, y, z);
+    }
+
+    /**
+     * Called upon the block being destroyed by an explosion
+     */
+    public void onBlockDestroyedByExplosion(World worldIn, int x, int y, int z, Explosion explosionIn) {
+        stopSinging(worldIn, x, y, z);
+    }
+
+    /**
+     * Called upon block activation (right click on the block.)
+     */
+    public boolean onBlockActivated(World worldIn, int x, int y, int z, EntityPlayer player, int side, float subX,
+            float subY, float subZ) {
+        if (super.onBlockActivated(worldIn, x, y, z, player, side, subX, subY, subZ)) {
+            stopSinging(worldIn, x, y, z);
+            return true;
+        }
+        return false;
+    }
+
+    public void stopSinging(World worldIn, int x, int y, int z) {
+        ChunkCoordinates chunkcoordinates = new ChunkCoordinates(x, y, z);
+        Minecraft mc = Minecraft.getMinecraft();
+        ISound isound = (ISound) mc.renderGlobal.mapSoundPositions.get(chunkcoordinates);
+
+        while (isound != null) {
```

Comment:
Both Minecraft and ISound are client-only classes that are being called server side here.
See https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16308 & 
[fml-server-latest.log](https://github.com/GTNewHorizons/twilightforest/files/15366762/fml-server-latest.log)